core/src/main/java/org/elasticsearch/index/query/TypeQueryBuilder.java

```diff
@@ -19,24 +19,48 @@
 
 package org.elasticsearch.index.query;
 
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.lucene.BytesRefs;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.index.mapper.DocumentMapper;
+import org.elasticsearch.index.mapper.internal.TypeFieldMapper;
 
 import java.io.IOException;
+import java.util.Objects;
 
 public class TypeQueryBuilder extends AbstractQueryBuilder<TypeQueryBuilder> {
 
     public static final String NAME = "type";
-    private final String type;
-    static final TypeQueryBuilder PROTOTYPE = new TypeQueryBuilder(null);
+
+    private final BytesRef type;
+
+    static final TypeQueryBuilder PROTOTYPE = new TypeQueryBuilder();
 
     public TypeQueryBuilder(String type) {
+        this.type = BytesRefs.toBytesRef(type);
+    }
+
+    public TypeQueryBuilder(BytesRef type) {
         this.type = type;
     }
 
+    public TypeQueryBuilder() {
```

Comment:
can we make this constructor private? null should not be an allowed value from the outside right?